store/postgres/tests/store.rs

```diff
@@ -1570,6 +1586,7 @@ fn entity_changes_are_fired_and_forwarded_to_subscriptions() {
         subscription
             .take(4)
             .collect()
+            .timeout(Duration::from_secs(3))
```

Comment:
This will help the test die more quickly if this comes up again instead of forcing Travis to time out (which takes 10 minutes)

Reply:
Alright, this is good enough for now.

Reply:
Oh, I also fixed the race condition, don't worry :)

Reply:
:joy: The store mutex guard lifetime wasn't long enough -- I had to move it outside of the `block_on` closure